src/block/StructureVoid.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\StructureVoidType;
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Vector3;
+
+class StructureVoid extends Transparent{
+	private StructureVoidType $type = StructureVoidType::VOID;
+
+	public function describeBlockItemState(RuntimeDataDescriber $w) : void{
+		$w->enum($this->type);
+	}
+
+	public function getType() : StructureVoidType{
+		return $this->type;
+	}
+
+	/** @return $this */
+	public function setType(StructureVoidType $type) : self{
+		$this->type = $type;
+		return $this;
+	}
+
+	/**
+	 * @return AxisAlignedBB[]
+	 */
+	protected function recalculateCollisionBoxes() : array{
+		return [];
+	}
+
+	public function isSolid() : bool{
+		return false;
+	}
+
+	public function canBeReplaced() : bool{
+		return true;
+	}
+
+	public function canBePlacedAt(Block $blockReplace, Vector3 $clickVector, int $face, bool $isClickedBlock) : bool{
+		return $blockReplace->getTypeId() !== BlockTypeIds::STRUCTURE_VOID && $blockReplace->canBeReplaced();
```

Comment:
```suggestion
		return !$blockReplace->hasSameTypeId($this) && $blockReplace->canBeReplaced();
```

Reply:
Why is this marked as resolved?

Reply:
I fixed it locally but have yet to push, I've been pretty busy IRL so this PR (and especially the structure block one) has to wait. I've pushed it now :)